src/components/cuda_pw/cupti_profiler.c

```diff
@@ -0,0 +1,1112 @@
+        .maxNumRanges = 1,
+        .maxNumRangeTreeNodes = 1,
+        .maxRangeNameLength = 64,
+    };
+
+    CUpti_Profiler_CounterDataImage_CalculateSize_Params calculateSizeParams = {
+        .structSize = CUpti_Profiler_CounterDataImage_CalculateSize_Params_STRUCT_SIZE,
+        .pPriv = NULL,
+        .sizeofCounterDataImageOptions = CUpti_Profiler_CounterDataImageOptions_STRUCT_SIZE,
+        .pOptions = &counterDataImageOptions,
+    };
+    CUPTI_CALL(cuptiProfilerCounterDataImageCalculateSizePtr(&calculateSizeParams), return PAPI_ENOSUPP);
+
+    CUpti_Profiler_CounterDataImage_Initialize_Params initializeParams = {
+        .structSize = CUpti_Profiler_CounterDataImage_Initialize_Params_STRUCT_SIZE,
+        .pPriv = NULL,
+        .sizeofCounterDataImageOptions = CUpti_Profiler_CounterDataImageOptions_STRUCT_SIZE,
+        .pOptions = &counterDataImageOptions,
+        .counterDataImageSize = calculateSizeParams.counterDataImageSize,
+    };
+
+    ctl->counterDataImage.size = calculateSizeParams.counterDataImageSize;
+    ctl->counterDataImage.data = (uint8_t *) calloc(ctl->counterDataImage.size, sizeof(uint8_t));
+    if (ctl->counterDataImage.data == NULL) {
+        fprintf(stderr, "MEMORY ERROR: calloc ctl->counterDataImage.data failed!\n");
+        return PAPI_ENOMEM;
+    }
+
+    initializeParams.pCounterDataImage = ctl->counterDataImage.data;
+    CUPTI_CALL(cuptiProfilerCounterDataImageInitializePtr(&initializeParams), return PAPI_ENOSUPP);
+
+    CUpti_Profiler_CounterDataImage_CalculateScratchBufferSize_Params scratchBufferSizeParams = {
+        .structSize = CUpti_Profiler_CounterDataImage_CalculateScratchBufferSize_Params_STRUCT_SIZE,
+        .pPriv = NULL,
+        .counterDataImageSize = calculateSizeParams.counterDataImageSize,
+        .pCounterDataImage = initializeParams.pCounterDataImage,
+    };
+    CUPTI_CALL(cuptiProfilerCounterDataImageCalculateScratchBufferSizePtr(&scratchBufferSizeParams), return PAPI_ENOSUPP);
+
+    ctl->counterDataScratchBuffer.size = scratchBufferSizeParams.counterDataScratchBufferSize;
+    ctl->counterDataScratchBuffer.data = (uint8_t *) calloc(ctl->counterDataScratchBuffer.size, sizeof(uint8_t));
+
+    CUpti_Profiler_CounterDataImage_InitializeScratchBuffer_Params initScratchBufferParams = {
+        .structSize = CUpti_Profiler_CounterDataImage_InitializeScratchBuffer_Params_STRUCT_SIZE,
+        .pPriv = NULL,
+        .counterDataImageSize = calculateSizeParams.counterDataImageSize,
+        .pCounterDataImage = initializeParams.pCounterDataImage,
+        .counterDataScratchBufferSize = ctl->counterDataScratchBuffer.size,
+        .pCounterDataScratchBuffer = ctl->counterDataScratchBuffer.data,
+    };
+    CUPTI_CALL(cuptiProfilerCounterDataImageInitializeScratchBufferPtr(&initScratchBufferParams), return PAPI_ENOSUPP);
+
+    return PAPI_OK;
+}
+
+static int reset_cuda_config_images(struct cupti_gpu_control_s *ctl)
```

Comment:
Shouldn't this be `reset_cupti_prof_config_images`?

Reply:
done.